network/src/validator_network/network_builder.rs

```diff
@@ -125,21 +118,14 @@ pub struct NetworkBuilder {
     connection_reqs_tx: libra_channel::Sender<PeerId, ConnectionRequest>,
     connection_reqs_rx: libra_channel::Receiver<PeerId, ConnectionRequest>,
     conn_mgr_reqs_tx: Option<channel::Sender<ConnectivityRequest>>,
-    connectivity_check_interval_ms: u64,
     max_concurrent_network_reqs: usize,
     max_concurrent_network_notifs: usize,
     max_connection_delay_ms: u64,
 }
 
 impl NetworkBuilder {
     /// Return a new NetworkBuilder initialized with default configuration values.
-    pub fn new(
-        executor: Handle,
-        network_id: NetworkId,
-        peer_id: PeerId,
-        role: RoleType,
-        listen_address: NetworkAddress,
-    ) -> NetworkBuilder {
+    pub fn new(network_config: NetworkConfig, executor: Handle, role: RoleType) -> NetworkBuilder {
```

Comment:
This change is going sideways at present.  Intuitively passing `network_config` as a parameter here rather than the individual subfields is an improvement because it reduces the number of parameters to `new(...)`.   But it's pyrrhic:  specifically, we (a) still have an inordinate number of fields in NetworkBuilder, (b) we're using network_config as a dictionary where we are interested in a very small number of the fields of network_config, and now (c) there are very similar fields which are in different places for seemingly arbitrary reasons.

The main pain point this PR is addressing is that there are too many fields in network_builder.  The 'fix' here  reduces the number of fields from way way too many to way too many, and introduces a struct field (`network_config`) with many unused subfields.  Not clear if that's a win.

Is there anyway that we can better structure the fields in `network_builder` so it is clear which component requires them?   Would it make sense for new to create a set of XXX configs, one per internal component, which are directly fed to those components?

Reply:
I'm not sure this is sideways. From your description it seems that it's an incremental improvement. I think to totally refactor the NetworkBuilder might be too much for one PR, and would potentially step on your toes even more. Right now even with this small change it seems that it would likely cause some merge conflict with your changes.

Reply:
> In #3342 they want to combine `NetworkConfig` and `NetworkBuilder`, and I think this is in the right direction to achieve that.

Can you sketch the next couple of steps to get there?  I know I've had challenges with incremental changes in the past when the reviewers didn't have a sense for the roadmap I was following.

Reply:
I don't have any roadmap for how to get the `NetworkBuilder` completely refactored...